index.js

```diff
@@ -2,33 +2,35 @@
 
 /* eslint-disable no-underscore-dangle */
 class NotificationBase {
-
     /**
      * Constructor for Build Notification
      * @method constructor
      * @param {Object}      config              Configuration specific to the plugin
-     * @param {Object}      server              API server
-     * @param {String}      eventName           Event name the notification plugin listen to
      * @return {NotificationBase}
      */
-    constructor(config, server, eventName) {
+    constructor(config) {
         this.config = config;
-        this.server = server;
-        this.eventName = eventName;
+    }
+
+    /**
+     * Get event list for receiving Screwdriver API events
+     * @method events
+     * @return {Array}
+     */
+    get events() {
+        return ['build_status'];
     }
 
     /**
      * Listen to the event and notify users based on their job settings
      * @method notify
-     * @return {Promise}
      */
-    notify() {
-        return this._notify();
+    notify(buildData) {
+        return this._notify(buildData);
```

Comment:
I think the logging an error in notifications-base makes the console noisy. For example, 
when `local.yaml` has some notification config but user's `screwdriver.yaml` does not have, the event sent to notification plugin and Joi.attempt() throws an validation error like [this line](https://github.com/screwdriver-cd/notifications-email/pull/10/files#diff-168726dbe96b3ce427e7fedce31bb0bcR72).